File: examples_new/player_random-walk_py/player_random-walk.py
#!/usr/bin/python3

# Author(s): Luiz Felipe Vecchietti, Chansol Hong, Inbae Jeong
# Maintainer: Chansol Hong (cshong@rit.kaist.ac.kr)

import random
import os
import sys
sys.path.append(os.path.dirname(os.path.realpath(__file__)) + '/../common')
try:
    print(sys.path)
    from participant import Participant
except ImportError as err:
    print('player_random-walk: \'participant\' module cannot be imported:', err)
    raise


class RandomWalkPlayer(Participant):
    def init(self, info):
        self.number_of_robots = info['number_of_robots']
        self.max_linear_velocity = info['max_linear_velocity']

    def update(self, frame):
        speeds = []
        for i in range(self.number_of_robots):
            speeds.append(random.uniform(-self.max_linear_velocity[i], self.max_linear_velocity[i]))
            speeds.append(random.uniform(-self.max_linear_velocity[i], self.max_linear_velocity[i]))
        self.set_speeds(speeds)


if __name__ == '__main__':
    player = RandomWalkPlayer()
    player.run()

File: examples_new/player_random-walk_cpp/player_random-walk.cpp
#include "player.hpp"

class RandomWalkPlayer : public Player {

public:
  RandomWalkPlayer(char **argv) : Player(argv) {}
  virtual ~RandomWalkPlayer() {}

  void init(json info) override {
    mNumberOfRobots = info["number_of_robots"];
    for (int i = 0; i < mNumberOfRobots; ++i)
      mMaxSpeed.push_back(info["max_linear_velocity"][i]);
  }

  void update(json frame) override {
    std::vector<double> speeds;
    for (int i = 0; i < 2 * mNumberOfRobots; ++i)
      speeds.push_back(2.0 * mMaxSpeed[i / 2] *
                       (0.5 - (double)rand() / RAND_MAX));
    setSpeeds(speeds);
  }

private:
  std::vector<double> mMaxSpeed;
  int mNumberOfRobots;
};

int main(int argc, char **argv) {
  RandomWalkPlayer *player = new RandomWalkPlayer(argv);
  player->run();
  delete player;
  return 0;
}
